### ISSUE TITLE
CLI tests

### ISSUE THREAD

**Issue by bkappes** [csm-adapt/confluence#11]
Please add tests of the anticipated CLI use cases, e.g.

```
import subprocess
...
def test_cli(...):
    commands = [
        "confluence merge -o test_files/Newfile.xlsx test_files/complete_excel.xlsx",
        "confluence merge -o test_files/Newfile2.xlsx test_files/partial_excel1.xlsx test_files/partial_excel2.xlsx"
    ]
    for cmd in commands:
        rval = subprocess.call(cmd, shell=True)
        assert rval == 0, f"ERROR: {cmd} failed."
```

**Comment by connorweddle2** [comment]
Should the CLI tests be added within the test_merge.py module, or should it be a separate module?

**Comment by bkappes** [comment]
I would leave it in test_merge, since

“confluence merge …”

will call merge.py, I assume? I assume

[…]$ confluence [confluence options] merge [merge options] file1 [file2 [file3 […]]]

calls  “confluence.py” or “cli.py”, which handles the confluence-specific options and then calls “merge.py” and that “merge.py” handles the remaining options and filenames.

Is the correct?

Best,

Branden

**Comment by connorweddle2** [comment]
I don't believe the current implementation utilizes any confluence-specific options, or at least I haven't used any as of yet. Perhaps @amikulichmines has some insight on that. It is currently run as...

[…]$ python merge.py [merge options] file1 file2 file3

I've added CLI tests to the test_merge.py module using the subprocess library. Instead of subprocess.call() I used subprocess.popen() so that I could change the working directory of the subprocess to merge.py.

Let me know if this works for the implementation, or if it needs to be adjusted!

**Comment by bkappes** [comment]
We’ll need to include an entry point (confluence.py) in the main source directory that handles confluence-specific command line options then calls merge.py.

Ideally the logic behind merge operations should be separated from the logic behind using merge. You have both done a great job keeping those separate, such that “run” is the entry point into the merge operations.

The logic behind how the user has asked for changes to merge operations should be separated, e.g. confluence-merge.py.

Let me know if you would like to meet and discuss.

Best,

Branden

**Comment by connorweddle2** [comment]
Dr. Kappes,

Thanks for the direction! I think a meeting to discuss the confluence-merge logic would be very helpful. 

-Connor

**Comment by bkappes** [comment]
I just wanted to make sure we have a record of the discussion from earlier today and hit the high points.

1. Add `confluence.py` that handles the CLI logic of `confluence`.
2. Add `confluence-merge.py` that handles the logic of handling the CLI of `confluence merge`
3. Likely move `merge.run` to `confluence-merge.run`, but other than that, leave the merge functionality in `merge.py`.

**Comment by bkappes** [comment]
I took a quick look at the CLI tests and they are still calling `python merge.py...`. Based on our last meeting these should be migrated to `confluence merge ...`, `confluence list...`, etc.

If you are still having trouble getting that to work, we should meet again to follow up.

**Comment by amikulichmines** [comment]
Hello,

Connor and I met this morning to go over it, and we are in the process of
rewriting the test_merge.py function. Our issue was that we had to create a
separate environment for it to be able to accept 'confluence merge ...' as
an argument in the terminal, but we have since fixed it using the
'setup.py' file. The test_merge should be fixed shortly, as well as the PIF
function.

Thank you,
Alex

**Comment by bkappes** [comment]
I’m glad you got that working.
